File: src/main/java/at/asit/terminal/prototype/server/WebSecurityConfiguration.java
package at.asit.terminal.prototype.server;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.web.SecurityFilterChain;
import org.springframework.security.web.authentication.AuthenticationFailureHandler;

@Configuration
public class WebSecurityConfiguration {

    @Bean
    public SecurityFilterChain customerTerminalAuthentication(HttpSecurity http) throws Exception {
        http.authorizeHttpRequests()
                .requestMatchers("/terminal")
                .authenticated().and().oauth2Login();
//        siopv2Login(http.authorizeHttpRequests()
//                .requestMatchers("/terminal/wallet")
//                .authenticated().and())
//                .failureHandler(new RedirectingAuthenticationFailureHandler());
//                .permitAll();
        http.authorizeHttpRequests()
                .anyRequest()
                .permitAll();
//                .authenticated().and().oauth2Login(); // employee login configuration
        return http.build();
    }
}

File: src/main/kotlin/at/asit/apps/terminal_sp/prototype/server/WebSecurityConfiguration.kt
package at.asit.apps.terminal_sp.prototype.server

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.web.SecurityFilterChain

@Configuration
class WebSecurityConfiguration {
    @Bean
    fun secretPageAuthentication(http: HttpSecurity): SecurityFilterChain {
        http.oauth2Login {
//            it.apply {
//                loginPage("/login/oauth2")
//            }
        }.authorizeHttpRequests {
            it.apply {
                requestMatchers("/terminal/oauth2").authenticated()
            }
//        }.siop2Login {
//            it.apply {
//                loginPage("/login/siop2")
//            }
//        }.authorizeHttpRequests {
//            it.apply {
//                requestMatchers("/siop2Page").authenticated()
//            }
        }.authorizeHttpRequests {
            it.apply {
                anyRequest().permitAll()
            }
        }
        return http.build()
    }
}